apps/server/src/apollo/resolvers/Query/posts.ts

```diff
@@ -1,6 +1,23 @@
+import type { QueryPostsArgs } from "@src/__generated__/graphql";
 import type { Context } from "@src/server/context";
 
-export const posts = async (_root: unknown, _args: never, ctx: Context) => {
+export const posts = async (
+  _root: unknown,
+  args: QueryPostsArgs,
+  ctx: Context
+) => {
+  if (args.options && args.options.beds) {
```

Comment:
I think we should rename `beds` to `availableBeds` here just to reduce confusion

Reply:
And in the other places as well

Reply:
on it